client/driver/mock_driver_testing.go

```diff
@@ -1,4 +1,4 @@
-// +build nomad_test
+// +build !release
```

Comment:
Will this be enabled when we are just doing local manual testing? Would this matter?

Reply:
Updated the PR description with an explanation of why I did this; sorry for not doing that initially!

But yeah, this PR means `go install` and `make dev` would include the `mock_driver` so it could be used when testing.

Only `make (pre)release` would *not* enable it.